From 18da95d9731e472dce66da85e8bbebaa86a9c4b9 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Wed, 6 Dec 2017 22:05:31 +0800
Subject: [PATCH 2998/5242] MLK-17077 staging: typec: clear vbus change event
 in irq handler

commit  c5500a2ee4488adc0e1774475a2d8ea410066336 from
https://source.codeaurora.org/external/imx/linux-imx.git

For vbus change event, we need read the vbus status to clear
the alert. Current code do this in queue work, this has problem
on single core running, the queue work of vbus change may have
no chance to be scheduled as we continue receive the vbus change
event in threaded irq.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index 4049b25..bfb649e 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -574,6 +574,9 @@ irqreturn_t tcpci_irq(struct tcpci *tcpci)
 	if (status & TCPC_ALERT_POWER_STATUS) {
 		unsigned int reg;
 
+		/* Read power status to clear the event */
+		regmap_read(tcpci->regmap, TCPC_POWER_STATUS, &reg);
+
 		regmap_read(tcpci->regmap, TCPC_POWER_STATUS_MASK, &reg);
 
 		/*
-- 
1.7.9.5

